### DISCUSSION TITLE
New script which restores the page actions menu

### DISCUSSION THREAD

**Discussion by Aleksandra0056** [Aris-t2/CustomJSforFx#84] [Ideas]
Hello, I would like a script which would restore the page actions menu, which was in the right of the address bar.

**Comment by Aleksandra0056** [comment]
Any news here? Why did nobody respond?

**Comment by Aris-t2** [comment]
Hi, code-wise the "boxes" for page action buttons are still present, even on Nightly builds, but add-ons do not add anything there anymore.

![image](https://github.com/Aris-t2/CustomJSforFx/assets/6013569/3d752c8e-52bd-41d7-8881-000494141196)
![image](https://github.com/Aris-t2/CustomJSforFx/assets/6013569/5c19d5de-742a-4d86-a2ff-ca7d576c24c6)

-------------------------------------

It required some manual work, but add-on buttons can be moved there:

![image](https://github.com/Aris-t2/CustomJSforFx/assets/6013569/daf8ff47-c961-45f7-a8a8-2588477704d4)

Add this to your `userChrome.css`:

```css
/* make the ... visible again */
#pageActionButton {
  visibility: visible !important;
}

/* disable browser default width for those buttons/icons */
.urlbar-page-action {
  width: auto !important; /* or unset */
}
```

Inside your script file you have to use code like this and add the ids for toolbar buttons you want to move to the address bar:

```js
(function() {

  document.getElementById("pageActionButton").appendChild(document.getElementById("ublock0_raymondhill_net-BAP"));
  document.getElementById("pageActionButton").appendChild(document.getElementById("firefox_ghostery_com-BAP"));
  document.getElementById("pageActionButton").appendChild(document.getElementById("_7a7a4a92-a2a0-41d1-9fd7-1e92480d612d_-BAP"));
  document.getElementById("pageActionButton").appendChild(document.getElementById("cookieautodelete_kennydo_com-BAP"));

}());
```

1. Move your buttons to navigation toolbar
2. Inspect buttons to get their ids using browsers dev tools
3. Add ids to script like seen in the above example
4. Clear cache / restart browser (might only work after 2-3 tries, but then it sticks)

**Reply by Aleksandra0056** [reply]
Thank you for quick response! And how can I make that extensions page actions will not show in address bar, but only in page actions menu? Because they are in address bar and also in page actions menu (duplicate items).

**Reply by Aleksandra0056** [reply]
And how can I re-add “Take a screenshot”, “Email link”, “Copy link” & “Send tab to device” items back to page actions menu?

**Reply by Aleksandra0056** [reply]
And how can I inspect the Firefox's UI? I opened the dev tools and I can't inspect the UI, but only the websites.

**Comment by Aris-t2** [comment]
The above was just a prove of concept that "something" can be done. It won't restore the old behavior.

Best solution might be just moving the required buttons to location bar without hiding them inside page actions sub menu:

```js
(function() {

  document.getElementById("page-action-buttons").insertBefore(document.getElementById("ublock0_raymondhill_net-browser-action"), document.getElementById("star-button-box"));
  document.getElementById("page-action-buttons").insertBefore(document.getElementById("screenshot-button"), document.getElementById("star-button-box"));
  document.getElementById("page-action-buttons").insertBefore(document.getElementById("email-link-button"), document.getElementById("star-button-box"));

}());
```

![image](https://github.com/Aris-t2/CustomJSforFx/assets/6013569/c3205731-dd70-4e20-b52c-35b411e34c19)


This will help showing how to find IDs and classes of UI items.
https://github.com/Aris-t2/CustomCSSforFx#how-to-find-item-ids-and-attributes

**Reply by Aleksandra0056** [reply]
Can you write solution for one of my questions: “Thank you for quick response! And how can I make that extensions page actions will not show in address bar, but only in page actions menu? Because they are in address bar and also in page actions menu (duplicate items).”

**Reply by Aris-t2** [reply]
Post screenshots, I don't know how you got duplicate items.

**Reply by Aleksandra0056** [reply]
No, no, no! I didn't changed anything to address bar's CSS, I didn't added any item to the address bar. I said, that EXTENSIONS ADDRESS BAR items are also in address bar and also in page actions menu (items like FlagFox, PWAsForFirefox, Multi-Account containers) these items are in address bar from that while when I've installed these extensions. I will not post any screenshots, because you didn't understood my question.

I want to know how can I keep these items in page actions menu and hide these items from address bar.

![Page actions problem](https://github.com/Aris-t2/CustomJSforFx/assets/69351811/5d4390de-a78e-409a-a4d3-3bdaf288cc2d)

**Reply by Aris-t2** [reply]
Well this was heavily misleading on so many levels... ALWAYS post screenshots to describe an issue. People are forced to guess otherwise and most likely wont respond.

--------------------

It seems your initial question

"Hello, I would like a script which would restore the page actions menu, which was in the right of the address bar."

was about, what is shown on your screenshot, and has nothing to do with this projects JavaScript scripts, right?


Anyways, hide items using CSS like in this example:

```css
#pageAction-urlbar-_1018e4d6-728f-4b20-ad56-37578a4de76b_ ,  /* Flagfox */
#pageAction-urlbar-_testpilot-containers,  /* Multi-Account containers */
#pageAction-urlbar-a2fff151f5ad0ef63cbd7e454e8907c1fa9cc32008f489178775570374f408a7_pokeinthe_io /* Certificate Viewer*/
{
  display: none !important;
}
```

(Only those show up for me on Fx 115+.)

You will have to use dev tools / browsers inspect feature to find all the ids you need to hide.

**Reply by Aleksandra0056** [reply]
> Well this was heavily misleading on so many levels... ALWAYS post screenshots to describe an issue. People are forced to guess otherwise and most likely wont respond.

I'm sorry, I had a bad mood yesterday.

Thank you! That code you posted helped me so much.

**Comment by Aleksandra0056** [comment]
Thank you all for your help! I no longer need this topic opened, so I'm closing this as solved because all my questions were answered.
